hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java

```diff
@@ -370,4 +373,27 @@ public void cancelUninterruptedTask_waitUntilRunCompleted_killMember_checkStatus
         assertTrue(future.isDone());
     }
 
+    @Test
+    public void reschedulingAfterMigration_whenCurrentNodePreviouslyOwnedTask() {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+        HazelcastInstance first = factory.newHazelcastInstance();
+
+        int tasksCount = 1000;
+        final IScheduledExecutorService scheduler = getScheduledExecutor(new HazelcastInstance[] {first }, "scheduler");
+        for (int i = 1; i <= tasksCount; i++) {
+            scheduler.scheduleAtFixedRate(named(valueOf(i), new EchoTask()), 5, 10, SECONDS);
+        }
+
+        assertTrueEventually(new AllTasksRunning(scheduler));
+
+        // Start a second member
+        HazelcastInstance second = factory.newHazelcastInstance();
+        waitAllForSafeState(first, second);
+
+        // Kill the second member, tasks should now get rescheduled back in first member
```

Comment:
How can you be sure the task was migrated to the second member in between? `assertTrueEventually(new AllTasksRunning(scheduler));` should also pass if there was no migration at all, shouldn't it?